src/unk_020655F4.c

```diff
@@ -19,68 +22,66 @@
 #include "unk_02005474.h"
 #include "unk_02069BE0.h"
 #include "unk_020711EC.h"
+#include "unk_020EDBAC.h"
 
-#include "constdata/const_020EE31C.h"
-#include "constdata/const_020EE4B8.h"
-
-typedef struct {
+typedef struct UnkStruct_020EEE54 {
     int unk_00;
     int unk_04;
     int unk_08;
     MapObject *unk_0C;
     const MapObjectAnimCmd *unk_10;
 } UnkStruct_020EEE54;
 
-typedef struct {
-    u16 unk_00;
-    s16 unk_02;
-    int unk_04;
-    fx32 unk_08;
-} UnkStruct_02065980;
+typedef struct WalkMovementData {
+    u16 unused;
+    s16 duration;
+    int dir;
+    fx32 distance;
+} WalkMovementData;
 
-typedef struct {
-    u16 unk_00;
-    s16 unk_02;
-} UnkStruct_02065C64;
+typedef struct WalkOnSpotMovementData {
+    u16 unused;
+    s16 duration;
+} WalkOnSpotMovementData;
 
-typedef struct {
-    fx32 unk_00;
+typedef struct JumpMovementData {
+    fx32 distance;
     fx32 unk_04;
     u16 unk_08;
     u16 unk_0A;
-    s8 unk_0C;
-    s8 unk_0D;
-    u8 unk_0E;
+    s8 dir;
+    s8 duration;
+    u8 unused;
     s8 unk_0F;
-} UnkStruct_02065E0C;
+} JumpMovementData;
 
-typedef struct {
-    int unk_00;
-} UnkStruct_020662AC;
+typedef struct DelayMovementData {
+    int delay;
+} DelayMovementData;
```

Comment:
**question**: Would the code look/feel cleaner if we decayed this `struct` to a `typedef`?

Reply:
Resolved on discord; we will keep this as a `struct`.